src/Cell.cpp

```diff
@@ -1056,24 +1088,19 @@ void Cell::ringify(std::vector<Cell*>* subcells) {
   ZCylinder* zcylinder = new ZCylinder(x1, y1, radius1);
   zcylinders.push_back(zcylinder);
 
-  /* Loop over ZCylinders and create a new Cell clone for each ring */
-  std::vector<ZCylinder*>::iterator iter2;
-  std::vector<Cell*>::iterator iter3;
-
+  /* Create ring Cells with successively smaller ZCylinders */
   for (iter2 = zcylinders.begin(); iter2 != zcylinders.end(); ++iter2) {
 
     /* Create ZCylinders for each of the sectorized Cells */
     if (subcells->size() != 0) {
       for (iter3 = subcells->begin(); iter3 != subcells->end(); ++iter3) {
-        log_printf(DEBUG, "Creating a new ring in sector Cell %d",
+        log_printf(DEBUG, "Creating a new ring in sector Cell ID=%d",
                    (*iter3)->getId());
 
         /* Create a new Cell clone */
         Cell* ring = (*iter3)->clone();
         ring->setNumSectors(0);
         ring->setNumRings(0);
-
-        /* Add new bounding ZCylinder surfaces to the clone */
         ring->addSurface(-1, (*iter2));
```

Comment:
If you implement the change above to hold the equal volume invariant you should make a conditional on adding this surface match the conditional on adding the outer cell above.